bookkeeper-http/http-server/src/main/java/org/apache/bookkeeper/http/AbstractHttpHandlerFactory.java

```diff
@@ -0,0 +1,135 @@
+/**
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+package org.apache.bookkeeper.http;
+
+/**
+ * Abstract handler factory which provide interface
+ * to create handlers for bookkeeper http endpoints.
+ */
+public abstract class AbstractHttpHandlerFactory<Handler> {
+    private HttpServiceProvider httpServiceProvider;
+
+    public AbstractHttpHandlerFactory(HttpServiceProvider httpServiceProvider) {
+        this.httpServiceProvider = httpServiceProvider;
+    }
+
+    public HttpServiceProvider getHttpServiceProvider() {
+        return httpServiceProvider;
+    }
+
+    /**
+     * Create a handler for heartbeat api.
+     */
+    public abstract Handler newHeartbeatHandler();
```

Comment:
Idea:
can you evaluate to have a single method
newHandler(enum ApiType)
we will have a more extensible API for the future

Reply:
Thanks. This is following the structure of @yzang's solid work in #278.  It seems be a good structure, with good abstract, and easy to extend for me. :) . Maybe we could change it later, when we found it is indeed not good extensible.
@yzang to comments on this idea.

Reply:
@zhaijack fine for me. I just wanted to know if you had evaluated alternative approches

Reply:
@eolivelli  That's a good idea, I think if we want more abstraction and simplicity, we can try that way

Reply:
Thanks. will do it in a more abstraction way.